### PR TITLE
[BUG FIX][ISSUE-513] Fixes pip install of kfserving

### LINKED ISSUE #513: pip install kfserving fails
/kind bug

**What steps did you take and what happened:**
```
> virtualenv venv -p python3
> source venv/bin/activate
> pip --version
pip 19.3.1 from /Users/ifilonenko/Projects/e2e-tests/venv/lib/python3.7/site-packages/pip (python 3.7)
> pip install kfserving==0.2.1
...
Successfully built tornado pyyaml future pycparser googleapis-common-protos
ERROR: azure 4.0.0 has requirement azure-storage-blob~=1.3, but you'll have azure-storage-blob 12.0.0 which is incompatible.
...
> python
```
```python
>>> import kfserving
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ifilonenko/Projects/e2e-tests/venv/lib/python3.7/site-packages/kfserving/__init__.py", line 18, in <module>
    from .storage import Storage
  File "/Users/ifilonenko/Projects/e2e-tests/venv/lib/python3.7/site-packages/kfserving/storage.py", line 22, in <module>
    from azure.storage.blob import BlockBlobService
ImportError: cannot import name 'BlockBlobService' from 'azure.storage.blob' (/Users/ifilonenko/Projects/e2e-tests/venv/lib/python3.7/site-packages/azure/storage/blob/__init__.py)
>>>
```

**What did you expect to happen:**

... no error

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version: N/A
- Knative Version: N/A
- KFServing Version: 0.2.1
- Kubeflow version: N/A
- Minikube version: N/A
- Kubernetes version: (use `kubectl version`): N/A
- OS (e.g. from `/etc/os-release`): N/A

### PR DESCRIPTION
**Pull request by ifilonenko**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #513

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/514)

### CODE REVIEW

**Comment by ifilonenko** [comment]
@googlebot I signed it!

**Comment by yuzisun** [comment]
omg! I have waited for your first PR for too long!

File: python/kfserving/requirements.txt [hunk 1]
```diff
@@ -12,3 +12,4 @@ adal >= 1.2.2
 table_logger >= 0.3.5
 numpy
 azure
+azure-storage-blob == 1.3.0
```
Comment (yuzisun) [inline python/kfserving/requirements.txt]: @rakelkar do we need the entire `azure` or just `azure-storage-blob` ?

Comment (ifilonenko) [reply python/kfserving/requirements.txt]: @rakelkar @ellis-bigelow can we pin a lot of these versions? its a bit dangerous to just leave these without versions and have them self-resolve to the most recent. It causes errors like #513 to come up

Comment (ellis-bigelow) [reply python/kfserving/requirements.txt]: Does this mean azure is making backwards incompat changes? =/

**Comment by jinchihe** [comment]
@ifilonenko The pkg is removed in the https://github.com/kubeflow/kfserving/issues/513 since that failed due to install from setup.py, would you minder to try install the last kfserving SDK version?
```
pip install -U kfserving
```
I think The kfserving SDK 0.2.3 will be installed, to see if have any error? The azure will also install the `azure-storage-blob`.

/hold

File: python/kfserving/requirements.txt (hunk 1 above)
Comment (rakelkar) [reply python/kfserving/requirements.txt]: Yes they did indeed make a breaking change :( 
I pinned like you did here and removed the dependency on azure as well here: https://github.com/kubeflow/kfserving/pull/518

**Comment by animeshsingh** [comment]
@ellis-bigelow what's the version of kfserving which will make in kubeflow 0.7 rc1? 

Also @jinchihe shouldn't the SDK version coincide match kfserving?

**Comment by jinchihe** [comment]
@animeshsingh Generally SDK will always work with kfserving since that's just basic function API, but your point make sense, we should check or refresh the SDK if needed before big version release.

**Comment by animeshsingh** [comment]
Thanks @jinchihe . It should work, but even the version number should be consistent. So we should continue the SDK be 0.2.0.X until 0.2.1 comes out, and then its 0.2.1

**Comment by yuzisun** [comment]
this should be fixed by https://github.com/kubeflow/kfserving/pull/518 now.

/close

**Pull request closed without merging**